docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -213,111 +181,135 @@ group key distribution
-    end note
+  Enclave1      -> Enclave1         :   Sig_Enclave1 <- sign <Enclave1_VK, Enclave2_VK, Chaincode_EK, encrypted_keys_message> with Enclave1_SK
+  note left
+    Enclave1 signs the key material
+    it distributes together with the
+    sender and intended recipient identity
+  end note
 
-  Enclave1      -> Enclave1         :   key_distribution_message <- <ECC1_S_PK, ECC2_s_PK, CC_E_PK, SIG_ECC1>
+  Enclave1      -> Enclave1         :   cckey_distribution_message <- <Enclave1_VK, Enclave2_VK, Chaincode_EK, Sig_Enclave1>
 
-  return encrypted_keys_message, key_distribution_message /'ECC1 -> Enclave1'/
-  return encrypted_keys_message, key_distribution_message /'ECC_Manager1 -> ECC1'/
-  return encrypted_keys_message, key_distribution_message /'Admin1 -> ECC_Manager1'/
+  return encrypted_cckeys_message, cckey_distribution_message /'ECC1 -> Enclave1'/
+  return encrypted_cckeys_message, cckey_distribution_message /'E_Manager1 -> ECC1'/
+  return encrypted_cckeys_message, cckey_distribution_message /'Admin1 -> E_Manager1'/
 
   note right Admin1
-    Key distribution is in progress. We need a means to transfer the message with
+    Chaincode key distribution is in progress. We need a means to transfer the message with
     the encrypted keys to the recipient enclave. The next step uses the ledger
     itself for this data transfer. This additionally allows to log the two phases
     of the key distribution: (1) the sender enclave produced the encrypted data blob;
     (2) the recipient enclave correctly received it. This can be useful when dealing
-    with enclaves of two different orgs.
+    with enclaves of two different orgs, to show that the expected actions have been taken.
   end note
 
-  group start-key-distribution endorsement
-    Admin1      -> ERCC1            ++: Start_KeyDist(encrypted_keys_message, key_distribution_message)
-    ERCC1       -> ERCC1            :   check ECC1_S_PK, ECC2_s_PK are registered
-    ERCC1       -> TLCC_Enclave1++: can_endorse(ECC1_S_PK)
-    return true
-    ERCC1       -> TLCC_Enclave1++: can_endorse(ECC2_S_PK)
+  group start-cckey-distribution endorsement
+    Admin1      -> ERCC1            ++: Start_KeyDist(encrypted_cckeys_message, cckey_distribution_message)
+    ERCC1       -> ERCC1            :   check Enclave1_VK's and Enclave2_VK's Credentials are registered
+    ERCC1       -> ERCC1            :   check Enclave1_Credentials and Enclave2_Credentials have same CC_Id
+    note over ERCC1
+        - we cannot check whether Enclave1 can endorse,
+            because the chaincode definition might have changed,
+            and the keys are now being distributed to a new enclave.
+        - similarly, we cannot check version, sequence, endorsement
+            policy fields of the chaincode definition.
+        - however, we can enforce that the recipient enclave (1) matches
+            with the current chaincode definition, and (2) can do useful work.
+    end note
```

Comment:
I think we can drop this as we defined `can_endorse` above. If we add further text, i would though suggest to put it _after_ can_endorse as we have done for similar notes above and maybe add a backwards reference to the definition.
```suggestion
    note over ERCC1
    end note
```